### PR TITLE
[JUJU-1906] Fix secret set

### PR DESCRIPTION
**Pull request by ycliuhw**
Currently, the `secret-set` command clears the secret content if no arg is provided.
This PR ensures we only set data if the data is provided in the `secret-set` command arg.

```console

$ juju exec --unit hello-kubecon/0  'uri=$(secret-add --owner=unit foo=bar); echo $uri;secret-set $uri --label=ccc;'
secret:ccv84seffbaqh15v1q30

$ juju show-secret ccv84seffbaqh15v1q30 --reveal
ccv84seffbaqh15v1q30:
  revision: 1
  owner: hello-kubecon/0
  label: ccc
  created: 2022-10-06T07:23:32Z
  updated: 2022-10-06T07:23:32Z
  content: {}  # empty content
```

Drive-by: surface permission denied error for secret get and delete k8s API calls.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

```console
$ juju exec --unit hello-kubecon/0  'uri=$(secret-add --owner=unit foo=bar); echo $uri; secret-set $uri --label=aaa;'
secret:ccvs28mffbamreliccg0

$ juju exec --unit hello-kubecon/0  'uri=$(secret-add foo=bar); echo $uri; secret-set $uri --label=bbb;'
secret:ccvs2buffbamreliccgg

$ juju show-secret ccvs28mffbamreliccg0 --reveal
ccvs28mffbamreliccg0:
  revision: 1
  owner: hello-kubecon/0
  label: aaa
  created: 2022-10-07T06:03:15Z
  updated: 2022-10-07T06:03:15Z
  content:
    foo: bar

$ juju show-secret ccvs2buffbamreliccgg --reveal
ccvs2buffbamreliccgg:
  revision: 1
  owner: hello-kubecon
  label: bbb
  created: 2022-10-07T06:03:29Z
  updated: 2022-10-07T06:03:29Z
  content:
    foo: bar

```

## Documentation changes

No

## Bug reference

No

### CODE REVIEW

File: worker/uniter/runner/jujuc/secret-add.go [hunk 1]
```diff
@@ -171,6 +171,9 @@ func (c *secretAddCommand) Run(ctx *cmd.Context) error {
 		ownerTag = names.NewUnitTag(unitName)
 	}
 	updateArgs := c.marshallArg()
+	if updateArgs.Value == nil {
```
Comment (wallyworld) [inline worker/uniter/runner/jujuc/secret-add.go:174]: `|| len(arg.Value.EncodedValues()) == 0` as well?

File: worker/uniter/runner/context/secrets.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func (s *secretsChangeRecorder) update(arg uniter.SecretUpdateArg) {
 		if arg.Description != nil {
 			c.Description = arg.Description
 		}
-		if arg.Value != nil {
+		if arg.Value != nil && len(arg.Value.EncodedValues()) > 0 {
```
Comment (wallyworld) [inline worker/uniter/runner/context/secrets.go:63]: maybe add an IsEmpty() helper?

**Review by wallyworld — APPROVED** [review]

File: worker/uniter/runner/context/secrets.go (hunk 2 above)
Comment (ycliuhw) [reply worker/uniter/runner/context/secrets.go:63]: yeah, agreed

**Pull request merged**